business/namespaces.go

```diff
@@ -105,6 +115,12 @@ func (in *NamespaceService) GetNamespace(namespace string) (*models.Namespace, e
 	promtimer := internalmetrics.GetGoFunctionMetric("business", "NamespaceService", "GetNamespace")
 	defer promtimer.ObserveNow(&err)
 
+	if kialiCache != nil {
+		if ns := kialiCache.GetNamespace(in.k8s.GetToken(), namespace); ns != nil {
+			return ns, nil
+		}
+	}
+
```

Comment:
What if cache is enabled but namespace was not found / outdated ? Shouldn't we put it back to cache below?

Reply:
Yeah, I relied on GetNamespaces(), but your comment is quite valid.
It should also trigger the cache refresh on this case.